text/0000-declarative-root.md

```diff
@@ -0,0 +1,276 @@
+
+```html
+<!-- Invalid -->
+<template lwc:preserve-comments>
+    <!-- Comment -->
+    <lwc:root></lwc:root>
+</template>
+```
+
+#### Contents
+
+The `<lwc:root>` element may not have any contents other than whitespace and comments:
+
+```html
+<!-- Valid -->
+<template>
+    <lwc:root>
+        
+    </lwc:root>
+</template>
+```
+
+```html
+<!-- Valid -->
+<template>
+    <lwc:root>
+        <!-- Comment -->
+    </lwc:root>
+</template>
+```
+
+However, in the case of `lwc:preserve-comments`, comments inside of `<lwc:root>` are not allowed:
+
+```html
+<!-- Invalid -->
+<template lwc:preserve-comments>
+    <lwc:root>
+        <!-- Comment -->
+    </lwc:root>
+</template>
+```
+
+This restriction is why `lwc:inner-html` is not supported on `<lwc:root>`.
+
+### Timing
+
+In terms of timing, any attributes added to or removed from the root element should follow the timing applied
+to elements inside of the template:
+
+```html
+<template>
+    <lwc:root class={clazz}></lwc:root>
+    <div class={clazz}></div>
+</template>
+```
+
+In other words, in the above example, if `clazz` changes, then the root element's `class` attribute should be changed
+in the same tick as when the `<div>`'s `class` changes.
+
+The above also applies to the timing of when event listeners are attached using `on*`.
+
+No guarantees are made about the ordering of changes made to `<lwc:root>` versus elements inside of the template.
+
+### Precedence
+
+Attributes and event listeners added using `<lwc:root>` may conflict with those added by the parent component:
+
+```html
+<!-- parent.html -->
+<template>
+    <x-child class="foo" 
+             data-foo="bar" 
+             onclick={onClick} 
+             lwc:spread={others}>
+    </x-child>
+</template>
+```
+
+```html
+<!-- child.html -->
+<template>
+    <lwc:root class="quux" 
+              data-foo="toto" 
+              onclick={onClick} 
+              lwc:spread={others}>
+    </lwc:root>
+</template>
+```
+
+In cases of conflict, the following rules apply:
+
+- `lwc:spread` precedence applies [as normal](https://github.com/salesforce/lwc-rfcs/pull/52) within the `<lwc:root>` element itself.
+- For event listeners such as `onclick`, both event listeners are attached.
+- For the `class` attribute, strings are concatenated with a single whitespace (`" "`) character (e.g. `"foo quux"` in the above example). No attempt is made to deduplicate duplicate strings.
+- For all other attributes, the parent overrides the child's attribute (e.g. `data-foo` would be `"bar"` in the above example).
+
+The reason for this is to allow a component to set defaults for its root's behavior, while still allowing consumers to override those defaults.
+
+## Drawbacks
+
+Implementing this feature does require additional complexity, and moves us further away from standard custom element syntax.
+
+## Alternatives
+
+### Calling it "host" instead of "root"

```

Comment:
BTW I am a bit agnostic on "root" versus "host" versus some other name we can think of.  "Root" is also kind of confusing because it might make you think of [shadow root](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot).

Reply:
> "Root" is also kind of confusing because it might make you think of [shadow root](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot).

It's the first thing that came to my mind. My 2 cents is that "host" makes more sense. It also aligns with the terminology used by other frameworks.

Reply:
Hmmm, OK. I am fine with changing it to "host." At least we are in good company with other frameworks.

Reply:
b6df56f